### PR TITLE
Fixed Python packages table

### PR DESCRIPTION
**Pull request by Rutwa189**
Solution to #4252 to enable users to provide a custom path for python packages.

### CODE REVIEW

**Comment by obelisk** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit 5278afaa3c44eafda7f1ec070355a0846af8f203 (Job results: [3173](https://jenkins.osquery.io/job/osqueryPullRequestOSX/3173/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:no_good: The commit 5278afaa3c44eafda7f1ec070355a0846af8f203 (Job results: [3634](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3634/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 5278afaa3c44eafda7f1ec070355a0846af8f203 (Job results: [7184](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7184/)) failed one or more tests (**Linux**).

**Comment by uptycs-nishant** [comment]
I think if no explicit directory is mentioned in the query then all the python packages should be searched at default locations such as '/usr/local/lib', '/usr/lib/' etc... and reported.

File: osquery/tables/system/python_packages.cpp [hunk 1]
```diff
@@ -29,15 +29,13 @@ namespace tables {
...
-
-/// Locations of site and dist packages.
 const std::set<std::string> kPythonPath = {
     "/usr/local/lib/python2.7/dist-packages/",
     "/usr/local/lib/python2.7/site-packages/",
     "/usr/lib/python2.7/dist-packages/",
     "/usr/lib/python2.7/site-packages/",
     "/Library/Python/2.7/site-packages/",
-};
+    };
```
Comment (fmanco) [inline osquery/tables/system/python_packages.cpp]: Whitespaces

File: osquery/tables/system/python_packages.cpp [hunk 2]
```diff
@@ -48,12 +46,12 @@ const std::string kWinPythonInstallKey =
 
 void genPackage(const std::string& path, Row& r) {
   std::string content;
-
   if (!readFile(path, content).ok()) {
     TLOG << "Cannot find info file: " << path;
     return;
   }
 
+
```
Comment (fmanco) [inline osquery/tables/system/python_packages.cpp]: nit: empty line.

File: osquery/tables/system/python_packages.cpp [hunk 3]
```diff
@@ -94,11 +92,13 @@ void genSiteDirectories(const std::string& site, QueryData& results) {
       auto path = directory + "/METADATA";
       genPackage(path, r);
     } else if (directory.find(".egg-info") != std::string::npos) {
-      auto path = directory + "/PKG-INFO";
+      auto path = directory + "PKG-INFO";
```
Comment (fmanco) [inline osquery/tables/system/python_packages.cpp]: Why did you remove the `/` here and not above?

File: osquery/tables/system/python_packages.cpp [hunk 4]
```diff
@@ -122,11 +122,23 @@ void genWinPythonPackages(const std::string& keyGlob, QueryData& results) {
 #endif
 }
 
+
```
Comment (fmanco) [inline osquery/tables/system/python_packages.cpp]: nit: empty lines.

File: osquery/tables/system/python_packages.cpp [hunk 5]
```diff
@@ -122,11 +122,23 @@ void genWinPythonPackages(const std::string& keyGlob, QueryData& results) {
...
+
+
 QueryData genPythonPackages(QueryContext& context) {
   QueryData results;
-
-  for (const auto& key : kPythonPath) {
-    genSiteDirectories(key, results);
+  std::set<std::string> paths;
```
Comment (fmanco) [inline osquery/tables/system/python_packages.cpp]: This code can be refactored to not repeat the `genSiteDirectories()` loop. Make `paths` be `kPythonPath` if no constraints and then have the `genSiteDirectories()` only once.

File: specs/python_packages.table [hunk 6]
```diff
@@ -6,7 +6,10 @@ schema([
     Column("summary", TEXT, "Package-supplied summary"),
     Column("author", TEXT, "Optional package author"),
     Column("license", TEXT, "License under which package is launched"),
-    Column("path", TEXT, "Path at which this module resides")
+    Column("path", TEXT, "Path at which this module resides"),
+    Column("directory", TEXT, "Directory where Python modules are located")
```
Comment (fmanco) [inline specs/python_packages.table]: nit: missing trailing comma

**Comment by fmanco** [comment]
@obelisk yeah I wondered the same, but tests are passing...

**Comment by osqueryer** [comment]
:no_good: The commit 69bbd79390a30efa11a735baf7c5324096ae67cd (Job results: [3651](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3651/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit c148f33115bfb535a5a3ebcc73e4f9cf84418e7e (Job results: [3653](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3653/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 66f1b0c9f4f5efda6e9edec95a500f39a3b5b64a (Job results: [3654](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3654/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 41d1c5555eddfb023887c3e14c18e8c7609063da (Job results: [3655](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3655/)) failed the code audit or documentation test.
